python/drned_xmnr/op/base_op.py

```diff
@@ -195,64 +214,75 @@ class Progressor(object):
     Progress messages come in chunks and need to be re-chunked into
     lines.  This class is not directly related to The Noon Universe.
     """
-    def __init__(self, action):
+    def __init__(self, action: 'ActionBase') -> None:
         self.buf = ""
         self.action = action
 
-    def progress(self, chunk):
+    def progress(self, chunk: str) -> None:
         lines = chunk.split('\n')
         lines[0] = self.buf + lines[0]
         for line in lines[:-1]:
             self.action.progress_msg(line)
         self.buf = lines[-1]
 
 
+ParamType = TypeVar('ParamType', str, int, None)
+CBRes = TypeVar('CBRes')
```

Comment:
cryptic name, maybe change to `CallbackResult`?

Reply:
The name is bad indeed; but since the type is unbounded, I'd better avoid using any name that might suggest there is any limit on the type; "fixing" with `T`.